granulate_utils/metrics/yarn/utils.py

```diff
@@ -165,10 +167,26 @@ def find_yarn_home_dir(*, logger: Union[logging.Logger, logging.LoggerAdapter])
     Find YARN home directory from command line arguments
     """
     logger.debug("looking for running YARN processes")
-    lines = subprocess.run(["ps", "-ax"], capture_output=True, text=True).stdout.split(" -D")
+
+    lines = subprocess.run(["ps", "-ax"], capture_output=True, text=True).stdout
```

Comment:
We can maybe filter only 'yarn' user processes but I'm not sure that this user is exists in every yarn environment

Reply:
Not sure it exists + not sure that in the context this code executes, we'll be able to resolve the numeric UID to the string UID `yarn`. So no need to do that.